[network] Fixed No IPv6 issue and rewrote check

elafarge opened the pull request:
In the first place, I just fixed a bug that was making gohai fail
completely when no IPv6 address was set on the Machine. I actually ended
up rewriting the whole network check for Windows because a couple of
things didn't make sense at all (for instance, we could end up sending
the Mac Address of a given interface, the IPv4 of another one and the
IPv6 of another again... I guess we absolutely need to avoid that in the
future).

I also added a test for my change. Run it with

```
go test *_windows.go *_windows_test.go
```

It was a real pain since I'm using the output of ipconfig on Windows for
that test but \x00 are appended after every character in my text (I am
sure it's an encoding issue but I didn't find a proper way to solve it
once and for all... apart from this hack).

NOTE: this has also been tested on a Windows Surface Pro I had at hand :)

remh commented on network/network_windows.go:
```diff
@@ -8,21 +8,48 @@ import (
+        // The line below is here in case we have to convert some Unicode to ASCII
+        // It shouldn't do anything on Windows but when running the tests (for
+        // Windows) on a Unix based-system, it's essential.
+        line = strings.Replace(line, "\x00", "", -1)
```
The indentation looks weird. Can you make sure to use go fmt please ?

elafarge commented:
@remh Sorry about the weird indentation, I forgot to run `go fmt` indeed. 

I fixed that and rebased.

For the future, I'll add a save-hook to my editor's configuration.

elafarge commented:
Thanks :)
